skimage/filters/__init__.py

```diff
@@ -22,50 +22,33 @@
 from ._unsharp_mask import unsharp_mask
 from ._window import window
 
+from .._shared import lazy
 
-__all__ = ['inverse',
-           'correlate_sparse',
-           'wiener',
-           'LPIFilter2D',
-           'gaussian',
-           'difference_of_gaussians',
-           'median',
-           'sobel',
-           'sobel_h',
-           'sobel_v',
-           'scharr',
-           'scharr_h',
-           'scharr_v',
-           'prewitt',
-           'prewitt_h',
-           'prewitt_v',
-           'roberts',
-           'roberts_pos_diag',
-           'roberts_neg_diag',
-           'laplace',
-           'farid',
-           'farid_h',
-           'farid_v',
-           'rank_order',
-           'gabor_kernel',
-           'gabor',
-           'try_all_threshold',
-           'meijering',
-           'sato',
-           'frangi',
-           'hessian',
-           'threshold_otsu',
-           'threshold_yen',
-           'threshold_isodata',
-           'threshold_li',
-           'threshold_local',
-           'threshold_minimum',
-           'threshold_mean',
-           'threshold_niblack',
-           'threshold_sauvola',
-           'threshold_triangle',
-           'threshold_multiotsu',
-           'apply_hysteresis_threshold',
-           'rank',
-           'unsharp_mask',
-           'window']
+__getattr__, __dir__, __all__ = lazy.install_lazy(
+    __name__,
+    submodules={'rank'},
+    submod_attrs={
```

Comment:
I see that specifying the attributes here seems to also have a desirable effect that when the user is in an interactive IPython session and does
```Python
skimage.filters.<TAB>
``` 
only the functions and not the submodule names show up in the list of autocompletions.

On the contrary, for `skimage.metrics`, I see the submodules `set_metrics` and `simple_metrics` show up in the list of autocompletion suggestions. Do you recommend extending this approach there or should we instead/also rename `set_metrics` -> `_set_metrics`, etc?

Reply:
I think this is an advantage of this approach, yes!  You can present whatever facade you want, despite how you implement things in the background.  Users were *never* meant to see our sub-submodules, and like you say the only way to suggest that to them before was to name them with an underscore.